lib/katello/tasks/fix_candlepin_product_content.rake

```diff
@@ -0,0 +1,43 @@
+namespace :katello do
+  desc "Fix missing repository content in candlepin"
+  task :fix_candlepin_content => ["environment", "check_ping"] do
+    logger = Logger.new(STDOUT)
+    User.current = User.anonymous_api_admin
+
+    Katello::Resources::Candlepin::Owner.all.each do |owner|
+      label = owner['key']
+      logger.debug("Checking Products in Organization #{owner['name']}(#{owner['key']})")
+
+      # find_missing associations
+      Katello::Resources::Candlepin::Product.all(owner['key']).each do |cp_product|
+        katello_product = Katello::Product.find_by_cp_id(cp_product['id'], Organization.find_by(label: owner['key']))
```

Comment:
I have moved the `Organization.find_by()` out of the loop. It might even be possible to split it up to get the Products for that organization and only here do the filter for `cp_id`.
I do not know if this would make it faster though :thinking: